### PR TITLE
Woo: Add and Delete Order Shipment Tracking

### LINKED ISSUE #1221: Woo: Add and delete shipment tracking records
Further implements the Shipment Tracking plugin by adding the ability to delete and existing shipment tracking record and add a shipment tracking record for an order through the API.

### PR DESCRIPTION
**Pull request by AmandaRiu**
Fixes #1221 

Adds the ability to add and delete order shipment tracking records for a single order. Also supports adding tracking with a custom provider.

## Automated Tests
Unit, Mocked and Release tests have also been added

### CODE REVIEW

File: example/src/main/java/org/wordpress/android/fluxc/example/WCAddOrderShipmentTrackingDialog.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            this.order = order
+            this.providers = providers
+        }
+    }
+
+    interface Listener {
+        fun onTrackingSubmitted(
+            site: SiteModel,
+            order: WCOrderModel,
+            tracking:
```
Comment (nbradbury) [inline example/src/main/java/org/wordpress/android/fluxc/example/WCAddOrderShipmentTrackingDialog.kt]: np: not sure why this is wrapped to two lines, but checkstyle doesn't seem to mind.

File: example/src/main/java/org/wordpress/android/fluxc/example/WCAddOrderShipmentTrackingDialog.kt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        val startDate = DateUtils.getCurrentDateString()
+        tracking_dateShipped.text = startDate
+        tracking_dateShipped.setOnClickListener {
+            displayDialog(
+                    DateUtils.getCalendarInstance(tracking_dateShipped.text.toString()), tracking_dateShipped)
+        }
+
+        tracking_ok.setOnClickListener {
+            val isCustomSelected = tracking_cboProvider.selectedItem == "Custom"
+            val tracking = WCOrderShipmentTrackingModel().apply {
```
Comment (nbradbury) [inline example/src/main/java/org/wordpress/android/fluxc/example/WCAddOrderShipmentTrackingDialog.kt]: The tracking number is being skipped here.

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderRestClient.kt [hunk 3]
```diff
@@ -394,6 +396,92 @@ class OrderRestClient(
...
+    ) {
+        val url = WOOCOMMERCE.orders.id(order.remoteOrderId).shipment_trackings.pathV2
+
+        val responseType = object : TypeToken<OrderShipmentTrackingApiResponse>() {}.type
+        val params = if (isCustomProvider) {
+            mutableMapOf(
+                    "custom_tracking_provider" to tracking.trackingProvider,
+                    "custom_tracking_link" to tracking.trackingLink)
+        } else {
+            mutableMapOf("tracking_provider" to tracking.trackingProvider)
```
Comment (nbradbury) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderRestClient.kt]: Is `tracking_link` being skipped here on purpose?

File: example/src/main/java/org/wordpress/android/fluxc/example/WCAddOrderShipmentTrackingDialog.kt (hunk 1 above)
Comment (AmandaRiu) [reply example/src/main/java/org/wordpress/android/fluxc/example/WCAddOrderShipmentTrackingDialog.kt]: Fixed in cbf62a85882924bd46143b2ead36ccbe4d9cfe80

File: example/src/main/java/org/wordpress/android/fluxc/example/WCAddOrderShipmentTrackingDialog.kt (hunk 2 above)
Comment (AmandaRiu) [reply example/src/main/java/org/wordpress/android/fluxc/example/WCAddOrderShipmentTrackingDialog.kt]: Good catch! Fixed in bda82417e16baabd69f76a04381eefede2296612

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderRestClient.kt (hunk 3 above)
Comment (AmandaRiu) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderRestClient.kt]: Yes, if using a standard provider, the tracking link is filled in on the backend by the extension.

**Comment by AmandaRiu** [comment]
@nbradbury All comments addressed. Ready for another round 👍

**Comment by nbradbury** [comment]
Looks good! :shipit:

**Pull request merged**